onmt/translate/translator.py

```diff
@@ -1079,7 +1079,9 @@ def _score_target(
         self, batch, memory_bank, src_lengths, src_vocabs, src_map
     ):
         tgt = batch.tgt
-        src = batch.src
+        src, src_lengths = (
+            batch.src if isinstance(batch.src, tuple) else (batch.src, None)
```

Comment:
Can you remind me the cases where we wouldn't have the src_lengths in the batch?

Reply:
I don't know when this could happen, I just copied the way of using batch.src in the code, it might be related to the torchtext TorchtextDataset part
It is not a good idea to have a variable that can be of two different types

Reply:
Indeed. Dates back to #1216 ...
I think this whole part will need some cleaning/refactoring at some point, e.g. when we'll look to put back word-feature support.